### DISCUSSION TITLE
Changing Username

### DISCUSSION THREAD

**Discussion by tracteurblinde** [FreshRSS/FreshRSS#5664] [Q&A] [answered]
Hello,

What is the correct way to change a user's username? I have an established user with data/history I want to preserve under the new name.

I see there is no official way to do this through the UI and I would have to move both files to disk as well as into SQLite. Is there already an established way to do this before I start hacking?

BR,
    - Tracteur Blindé

**Comment by Alkarex** [comment, marked as answer]
I do not believe we have documented that, but that depends on the database you are using.

1. If you use the default SQLite, you can just rename the folder from `./FreshRSS/data/users/bob/` to `./FreshRSS/data/users/alice/`
2. If you use another database (PostgreSQL, MariaDB, MySQL, or even SQLite), the safest option (besides manually renaming the correct SQL tables) is the following:
    * export the users' data to SQLite using our CLI
    * create a new user
    * import from SQLite into new user
    * copy the settings from `./FreshRSS/data/users/bob/config.php` to `./FreshRSS/data/users/alice/config.php`
    * delete old user
 
There are then a few details, such as existing API calls, WebSub, etc. For instance, for option 1, it is good (but will be automatically fixed over time), to rename `./data/PubSubHubbub/feeds/*/bob.txt` to `./data/PubSubHubbub/feeds/*/alice.txt`

**Comment by tracteurblinde** [comment]
Hello,

Thank you for the quick reply.

Turns out I had misremembered my configuration, it was postgres, but the comment still helped me.

1. Creating new user in the UI
2. Stop FreshRSS
3. User data copied from `./FreshRSS/data/users/bob/` to `./FreshRSS/data/users/alice/`
4. Individual tables copied from one user to another using postgres queries (i.e. `INSERT INTO alice_entry SELECT * FROM bob_entry;` )
5. Restart FreshRSS

By loading the UI I can log in using the created user and I see all the configuration/state/content I expect.

The only problem I encountered was that I had duplicate entries that Postgres was rightly complaining about inserting into the new table. (See [this comment](https://github.com/FreshRSS/FreshRSS/issues/5410#issuecomment-1703528403)). After removing duplicates and leaving only the original entry, I had no problems.

BR,
    - Tracteur Blindé

**Reply by tracteurblinde** [reply]
Hello,

This doesn't completely do the right thing for postgres because it does not update the sequences. You also need to update each sequence ala `SELECT setval('alice_feed_id_seq', (SELECT last_value from bob_feed_id_seq));` for each of the 3 sequences.

BR,
    - Tracteur Blindé

**Reply by Alkarex** [reply]
I think the export / import approach is safest and easiest, but this can also work :-)

**Comment by Elekam** [comment]
Hello, ~~I dont like reopening old issues, but~~ maybe this will help someone encountering this error aswell in the future.

Im using the sqlite database. I simply renamed the /user/username/ folder, but upon restarting the container I encounter a blank page with only the words "Error during context user init!".

The logs dont show anything of note, even with "enviroment=development" set, only 2 lines.
The new_username folder and all files inside have the permissions of root:www-data and chmod 0644.

To me this all looks correct, so I dont understand why it does not work.
I assume its a simple misunderstanding from my side. @Alkarex, any chance you could clear this up for me? If you need more info please tell me.

I can try the import/export approach, but Im curious to why renaming did not work when you said it should be as simple as that.

Additionally, I cant find the ./data/PubSubHubbub/feeds/*/username.txt. It only goes to ./data/PubSubHubbub/feeds/index.html, so I assume it is irrelevant here.

**Reply by Frenzie** [reply]
Did you try in a private window or after clearing cache? Maybe it doesn't interact well with old cookies.

**Reply by Elekam** [reply]
> Did you try in a private window or after clearing cache? Maybe it doesn't interact well with old cookies.

I tried right now, and am still getting that error :/
Very strange

**Reply by Elekam** [reply]
Aha, I found the problem. My old username was still set as the default user in config.php. After changing it to the new username, everything works as expected.

Im surprised though that I didnt find anything in the logs regarding this.
But problem solved, sorry for the bother and thanks!
